src/cli/commands/snip.ts

```diff
@@ -141,6 +147,23 @@ export const snip = async (args: SnipArgs): Promise<string[]> => {
         stateVersionWrittenForPath[document.path] = true;
       }
     }
+
+    if (id !== undefined) {
+      const idAttribute: string = document.attributes["snippet"];
+      if (typeof id === "string") {
+        if (idAttribute && idAttribute !== id) {
+          // Not the requested id
+          return;
+        }
+      } else {
+        if (idAttribute && !id.includes(idAttribute)) {
+          // Not the requested id
+          return;
+        }
```

Comment:
This if/else block looks like it could be cleaned up (usually the case if there is nothing but another `if` within an `if` block). Btw, check out `Array.isArray()`. You could probably write this condition like:

```
if (Array.isArray(id) ? !id.includes(idAttribute) : idAttribute !== id)
```

Another cute trick would be to always treat it as an array (`flat(1)` removes one 'level' of array in case id was already an array):

```
if (![id].flat(1).includes(idAttribute))
```

We can assert the idAttribute is not undefined, as it is _required_ by the snippet command so missing it would be a programmer error. (Your current implementation could hide this bug if it ever came to be.)

Reply:
Oooo that second conditional is super slick. Am using!